apps/prairielearn/src/ee/pages/lti13CourseNavigation/lti13CourseNavigation.ts

```diff
@@ -36,6 +33,57 @@ function prettyCourseName(ltiClaim) {
   }
 }
 
+async function courseInstancesAllowedToLink({
+  course_id,
+  resLocals,
+}: {
+  course_id: string;
+  resLocals: Record<string, any>;
+}) {
+  const course_instances = await selectCourseInstancesWithStaffAccess({
+    course_id,
+    user_id: resLocals.authn_user.user_id,
+    authn_user_id: resLocals.authn_user.user_id,
+    is_administrator: resLocals.authn_is_administrator,
+    authn_is_administrator: resLocals.authn_is_administrator,
+  });
+
+  return course_instances.filter((ci) => ci.has_course_instance_permission_edit);
+}
+
+async function coursesAllowedToLink({ resLocals }: { resLocals: Record<string, any> }) {
+  const courses = await selectCoursesWithEditAccess({
+    user_id: resLocals.authn_user.user_id,
+    is_administrator: resLocals.authn_is_administrator,
+  });
+
+  return courses.filter((c) => c.permissions_course.has_course_permission_edit);
+}
+
+router.get(
+  '/course_instances',
+  asyncHandler(async (req, res) => {
+    const course_instances = await courseInstancesAllowedToLink({
+      course_id: z.coerce.string().parse(req.query.unsafe_course_id),
```

Comment:
If this is in fact unsafe, the `unsafe_` prefix should propagate down to the `courseInstancesAllowedToLink` function (and the query) so we can verify that it's being validated correctly.

Reply:
Addressed in 098759ca by validating unsafe_course_id against the collection of safe courses for the user before passing into `courseInstancesAllowedToLink`